vignettes/col_plan.Rmd

```diff
@@ -14,6 +14,283 @@ knitr::opts_chunk$set(
+
+The specification and behavior of the `col_plan` is dependent on the number of column variables that are provided to the tfrmt:
+
+| col_plan options:        | 1 column variable | >1 column variable                |
+| ------------------------ | ----------------- | --------------------------------- |
+| **Rename columns**       | Yes               | Yes (lowest level col only)       |
+| **Reorder columns**      | Yes               | Yes                               |
+| **Add spanning labels**  | Yes               | No (provided by upper level cols) |
+
+As shown in the table, the `col_plan` is most powerful in conjunction with the use of a single `column` variable. However, in many cases, such as when a user performs the analysis by treatment group, these "by" variables can essentially be re-purposed as data-driven spanners in the form of upper level `column` variables. 
+
+Let's look at each of these key workflows in more detail.
+
+# Option 1: A single `column` variable 
+
+Let's consider the following dataset which includes a single `column` variable:
+
+```{r}
+dat <- tribble(
+    ~group,     ~label,  ~my_col,    ~parm, ~val,
+    "g1", "rowlabel1",  "col1"  ,  "value",    1,
+    "g1", "rowlabel1",  "col2"  ,  "value",    1,
+    "g1", "rowlabel1",  "mycol3",  "value",    1,
+    "g1", "rowlabel1",  "col4"  ,  "value",    1,
+    "g1", "rowlabel1",  "mycol5",  "value",    1,
+    "g1", "rowlabel2",  "col1"  ,  "value",    2,
+    "g1", "rowlabel2",  "col2"  ,  "value",    2,
+    "g1", "rowlabel2",  "mycol3",  "value",    2,
+    "g1", "rowlabel2",  "col4"  ,  "value",    2,
+    "g1", "rowlabel2",  "mycol5",  "value",    2,
+    "g2", "rowlabel3",  "col1"  ,  "value",    3,
+    "g2", "rowlabel3",  "col2"  ,  "value",    3,
+    "g2", "rowlabel3",  "mycol3",  "value",    3,
+    "g2", "rowlabel3",  "col4"  ,  "value",    3,
+    "g2", "rowlabel3",  "mycol5",  "value",    3)
+  
+```
+
+
+## No column spanners
+
+In the case of a single column variable with no column spanners, `col_plan` behaves identically to `dplyr::select`. 
+
+Let's say we want to remove one of the columns:
+```{r}
+tfrmt(
+    group = group,
+    label = label,
+    param = parm,
+    values = val,
+    column = my_col,
+    body_plan = body_plan(
+      frmt_structure(group_val = ".default", label_val = ".default", frmt("x"))
+    ),
+    col_plan = col_plan(
+      -mycol5            
+    )
+  ) %>% 
+  print_to_gt(dat)
+```
+
+Just like `dplyr::select`, we can also reorder, rename, and remove columns, using `tidyselect` syntax if we'd like:
+
+```{r}
+tfrmt(
+    group = group,
+    label = label,
+    param = parm,
+    values = val,
+    column = my_col,
+    body_plan = body_plan(
+      frmt_structure(group_val = ".default", label_val = ".default", frmt("x"))
+    ),
+    col_plan = col_plan(
+      group, 
+      label, 
+      starts_with("col"),
```

Comment:
Should we default to keeping these values when they are defined? I am feeling as though the users shouldn't need to define them. what are everyone's thoughts?

Reply:
I tend to agree. I like that people have control over order/dropping them, but i think it would be a nice default to keep them in if not specified.